components/library/src/content/page.rs

```diff
@@ -110,27 +104,47 @@ impl Page {
         let (meta, content) = split_page_content(file_path, content)?;
         let mut page = Page::new(file_path, meta, base_path);
 
-        page.lang = page.file.find_language(config)?;
+        if let Some(ref l) = page.file.maybe_lang {
+            if !config.is_multilingual() {
+                // TODO: add test
+                bail!(
+                    "Page `{}` has a language set, but `languages` in config.toml is empty. Note that file names must not contain dots.",
+                    file_path.display()
+                );
+            } else if l.is_empty() {
+                // TODO: add test
+                bail!("Page `{}` has an empty language. Did you mistakenly put 2 dots before the extension?", file_path.display());
+            } else if l == &config.default_language_options.language_alias {
+                bail!("Pages in the default language should not set it in their title (caused by `{}`).", file_path.display());
```

Comment:
does that cause any issues?

Reply:
Don't think so. The code currently does fail though if the default language is set in a file name.

```rust
if !config.languages_codes().contains(&parts[1].as_ref()) {
            bail!("File {:?} has a language code of {} which isn't present in the config.toml `languages`", self.path, parts[1]);
        }
```